retrievalmarket/impl/provider.go

```diff
@@ -92,13 +91,6 @@ func DealDeciderOpt(dd DealDecider) RetrievalProviderOption {
 	}
 }
 
-// DisableNewDeals disables setup for v1 deal protocols
```

Comment:
I'm not familiar with the v0/v1 deal proposal thing, but what are the implications of removing the ability to disable this? Is v0 ancient by now and we don't foresee any need to provide the optionality on opting in to v0-only at this stage?

Reply:
ah, I just re-read the OP:

> I'm also dropping all the v0 protocols, which haven't actually been in use since pre-mainnet.

so nevermind then; I see that v0 code is also being dropped completely with no migration path. Fair enough if it's as ancient as pre-mainnet.

Reply:
yea, I'm going to do an additional PR to drop the on disk storage portion as well... I have no idea why it lived this long.